source/common/network/address_impl.cc

```diff
@@ -212,17 +212,19 @@ bool Ipv4Instance::operator==(const Instance& rhs) const {
           (ip_.port() == rhs_casted->ip_.port()));
 }
 
-int Ipv4Instance::bind(int fd) const {
-  return ::bind(fd, reinterpret_cast<const sockaddr*>(&ip_.ipv4_.address_),
-                sizeof(ip_.ipv4_.address_));
+Result Ipv4Instance::bind(int fd) const {
+  return {::bind(fd, reinterpret_cast<const sockaddr*>(&ip_.ipv4_.address_),
+                 sizeof(ip_.ipv4_.address_)),
+          errno};
```

Comment:
It's quite possible that C++ guarantees that the ::bind call happens first, and then errno is captured, but TBH it's not very obvious to me that is the case. I would prefer that we do the call into a local variable and then return the result and errno. Same elsewhere. Thoughts?

Reply:
You're right. The [c++14 documentation](https://timsong-cpp.github.io/cppwp/n4140/dcl.fct.default#9) states:
> The order of evaluation of function arguments is unspecified.

I'll fix this.

Reply:
@venilnoronha You're referring to function arguments, but this is list-initialization, and C++14 guarantees the order:
https://timsong-cpp.github.io/cppwp/n4140/dcl.init.list#4

> Within the initializer-list of a braced-init-list, the initializer-clauses, including any that result from pack expansions ([temp.variadic]), are evaluated in the order in which they appear.

Reply:
@lizan my bad, I just realized that we switched over to a `struct`. Thanks for sharing!

I think I'd still have it explicit, at least for readability.

Reply:
Addressed in 7e05d55.